src/listeners.ts

```diff
@@ -0,0 +1,582 @@
+        class while in delete mode.
+        */
+        if (
+            old_id === DELETE_CLASS_ID &&
+            DELETE_MODES.includes(current_subtask.state.annotation_mode)
+        ) {
+            $("#toolbox_sel_" + DELETE_CLASS_ID).trigger("click");
+        }
+    }
+}
+
+/**
+ * Handler for ULabel keydown events.
+ *
+ * @param keydown_event Event to handle
+ * @param ulabel ULabel instance
+ * @returns Whether the event was handled
+ */
+function handle_keydown_event(
+    keydown_event: JQuery.KeyDownEvent,
+    ulabel: ULabel,
+): boolean {
+    const shift = keydown_event.shiftKey;
+    const ctrl = keydown_event.ctrlKey || keydown_event.metaKey;
+    const key_is_z = (
+        keydown_event.key === "z" ||
+        keydown_event.key === "Z" ||
+        keydown_event.code === "KeyZ"
+    );
+
+    if (ctrl && key_is_z) {
+        keydown_event.preventDefault();
+        if (shift) {
+            ulabel.redo();
+        } else {
+            ulabel.undo();
+        }
+        return false;
+    } else {
+        const current_subtask = ulabel.get_current_subtask();
+        switch (keydown_event.key) {
+            case "Escape":
+                // If in erase or brush mode, cancel the brush
+                if (current_subtask.state.is_in_erase_mode) {
+                    ulabel.toggle_erase_mode();
+                } else if (current_subtask.state.is_in_brush_mode) {
+                    ulabel.toggle_brush_mode();
+                } else if (current_subtask.state.starting_complex_polygon) {
+                    // If starting a complex polygon, undo
+                    ulabel.undo();
+                } else if (current_subtask.state.is_in_progress) {
+                    // If in the middle of drawing an annotation, cancel the annotation
+                    ulabel.cancel_annotation();
+                }
+                break;
+        }
+    }
+}
+
+/**
+ * Create listeners for a ULabel instance.
+ * Consider breaking out long handlers.
+ *
+ * @param ulabel ULabel instance
+ */
+export function create_ulabel_listeners(
+    ulabel: ULabel,
+) {
+    // ================= Mouse Events in the ID Dialog =================
+    const id_dialog = $(".id_dialog");
+    id_dialog.on(
+        "mousemove" + ULABEL_NAMESPACE,
+        function (mouse_event) {
+            if (!ulabel.get_current_subtask()["state"]["idd_thumbnail"]) {
+                ulabel.handle_id_dialog_hover(mouse_event);
+            }
+        },
+    );
+
+    // ================= Mouse Events in the Annotation Container =================
+    const annbox = $("#" + ulabel.config["annbox_id"]);
+
+    // Detect and record mousedown
+    annbox.on(
+        "mousedown" + ULABEL_NAMESPACE,
+        function (click_event) {
+            ulabel.handle_mouse_down(click_event);
+        },
+    );
+
+    // Prevent default for auxclick
+    $(document).on(
+        "auxclick" + ULABEL_NAMESPACE,
+        ulabel.handle_aux_click,
```

Comment:
This one is just a `.preventDefault()` call. Should I just remove that in favor of calling it here?

Reply:
eh, im ok with having the explicit handler even if it is just a placeholder for now